runtime/lua/vim/treesitter/query.lua

```diff
@@ -253,7 +253,9 @@ local directive_handlers = {
   ["set!"] = function(_, _, _, pred, metadata)
     if #pred == 4 then
       -- (#set! @capture "key" "value")
-      metadata[pred[2]][pred[3]] = pred[4]
+      local capture = pred[2]
+      metadata[capture] = {}
```

Comment:
This should do a check for a table already or else it will overwrite any other set directives for that table.

Reply:
Thanks for the catch, I've fixed it and added another test